network/src/peer/fuzzing.rs

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    constants::MAX_FRAME_SIZE,
+    interface::NetworkProvider,
+    peer::message_codec,
+    protocols::wire::{
+        handshake::v1::{MessagingProtocolVersion, SupportedProtocols},
+        messaging::v1::NetworkMessage,
+    },
+    testutils::fake_socket::ReadOnlyTestSocketVec,
+    transport::{Connection, ConnectionId, ConnectionMetadata},
+    ProtocolId,
+};
+use futures::{executor::block_on, future, io::AsyncReadExt, sink::SinkExt, stream::StreamExt};
+use libra_config::network_id::NetworkContext;
+use libra_logger::prelude::*;
+use libra_network_address::NetworkAddress;
+use libra_proptest_helpers::ValueGenerator;
+use libra_types::PeerId;
+use memsocket::MemorySocket;
+use netcore::{compat::IoCompat, transport::ConnectionOrigin};
+use proptest::{arbitrary::any, collection::vec};
+use std::time::Duration;
+use tokio_util::codec::FramedWrite;
+
+/// Generate a sequence of `NetworkMessage`, lcs serialize them, and write them
+/// out to a buffer using our length-prefixed message codec.
+pub fn generate_corpus(gen: &mut ValueGenerator) -> Vec<u8> {
+    let network_msgs = gen.generate(vec(any::<NetworkMessage>(), 1..20));
+
+    let network_msgs = network_msgs
+        .into_iter()
+        .map(|msg| lcs::to_bytes(&msg).unwrap())
+        .collect::<Vec<_>>();
+
+    let (write_socket, mut read_socket) = MemorySocket::new_pair();
+    let mut writer = FramedWrite::new(IoCompat::new(write_socket), message_codec(MAX_FRAME_SIZE));
```

Comment:
how will this change the data? add 2-byte length prefix?

Reply:
it writes each serialized `NetworkMessage` as a frame like `{ 4-byte length prefix } || { serialized network message }` into the io stream. the `MAX_FRAME_SIZE` actually limits the size of each serialized network message. so if we try to read or write a serialized network message to the io stream with length > MAX_FRAME_SIZE, it'll return an Err

Reply:
ah I see that [in the spec](https://github.com/libra/libra/blob/master/specifications/network/messaging-v1.md#framing), we might want to test that in combo with the noise fragmentation as well : o